acceptance-tests/src/main/java/io/blueocean/ath/SmartWebElement.java

```diff
@@ -73,23 +102,66 @@ public void forEach(java.util.function.Consumer<WebElement> fn) {
         }
     }
 
+    /**
+     * Determines if the element is visible
+     * @return true if visible, false if not
+     */
     public boolean isVisible() {
         return getElement().isDisplayed();
     }
 
+    /**
+     * Determines if the element is present
+     * @return true if present, false if not
+     */
+    public boolean isPresent() {
+        try {
+            if (getDriver().findElement(by) == null) {
+                return false;
+            }
+            return true;
+        } catch(NoSuchElementException e) {
+            return false;
+        }
+    }
+
     @Override
     public void click() {
-        forEach(e -> e.click());
+        for (int i = 0; i < RETRY_COUNT; i++) {
+            try {
+                WebElement e = getElement();
+                e.click();
+                if (i > 0) {
+                    logger.info(String.format("retry click successful for %s", by.toString()));
+                }
+                return;
+            } catch (WebDriverException ex) {
+                if (ex.getMessage().contains("is not clickable at point")) {
+                    logger.warn(String.format("%s not clickable: will retry click", by.toString()));
+                    logger.debug("exception: " + ex.getMessage());
+                    try {
+                        // typically this is during an animation, which should not take long
+                        Thread.sleep(500);
```

Comment:
I had assumed I'd need to add a sleep to this method, although it worked very reliably without it (at least in my local env). I suppose there's no harm in being more conservative by having it pause longer before retry.

Reply:
Well, what I'd like to do (and maybe what I did before, I don't quite remember) is get rid of the WebDriver `wait` and replace it with some smart polling / retries of our own. It should speed things up in some cases. We'll want to add some other utilities to try to have some explicit waits for a css selector, for example waiting for github to scan repos and such, we know that will take a while.